src/main/scala/sbtdatabricks/DatabricksPlugin.scala

```diff
@@ -183,8 +189,53 @@ object DatabricksPlugin extends AutoPlugin {
     }
   }
 
+  private lazy val executeCommandImpl: Def.Initialize[Task[Seq[CommandStatus]]] = Def.taskDyn {
+    Def.task {
```

Comment:
You don't need `Def.taskDyn` above. That's used when there is an if statement somewhere and what you need to do next depends on that if statement. Otherwise sbt will just try to do everything in parallel, which leads to race conditions (without the `taskDyn` in `dbcDeploy`, you get something like: upload jars -> restart clusters -> delete the jars).

I think it should behave properly below.